### PR TITLE
Remove sanitizing condition which does not prevent vulnerability.

### PR DESCRIPTION
**Pull request by pwntester**
By using a `Constructor` SnakeYAML allows to specify a root type which is actually used. Nested properties, however, are not type checked and therefore they can be used to embed the payload

### CODE REVIEW

**Comment by aschackmull** [comment]
```
[ERROR] Input file ql/java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll is not correctly formatted
```

File: java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll [hunk 1]
```diff
@@ -24,14 +17,11 @@ class SnakeYamlSafeConstructor extends RefType {
 }
 
 /**
- * An instance of `SafeConstructor` or a `Constructor` that only allows the type that is passed into its argument.
+ * An instance of `SafeConstructor`
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/SnakeYaml.qll:20]:
```suggestion
 * An instance of `SafeConstructor`.
```

**Comment by aschackmull** [comment]
```
Error:     ql/java/ql/test/query-tests/security/CWE-502/UnsafeDeserialization.qlref: Test failure (Expected output does not match)
```

**Review by aschackmull — COMMENTED** [review]
LGTM. Is this something for which we'll want a change-note or do you think it is too corner-case to warrant attention?

**Comment by pwntester** [comment]
Added a change note since that bug was causing some real FNs

**Review by aschackmull — APPROVED** [review]

**Pull request merged**